x/mongo/driver/topology/pool.go

```diff
@@ -596,6 +596,8 @@ func (p *pool) checkInNoEvent(conn *connection) error {
 		return ErrWrongPool
 	}
 
+	conn.bumpIdleDeadline()
```

Comment:
Maybe a comment here explaining why we bump the deadline?

Reply:
Good idea, will add.

Reply:
Added.